akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpResponseParser.scala

```diff
@@ -180,41 +180,37 @@ private[http] class HttpResponseParser(protected val settings: ParserSettings, p
         }
         case HttpMethods.CONNECT =>
           finishEmptyResponse()
-        case _ => teh match {
-          case None => clh match {
-            case Some(`Content-Length`(contentLength)) =>
-              if (contentLength == 0) finishEmptyResponse()
-              else if (contentLength <= input.size - bodyStart) {
-                val cl = contentLength.toInt
-                emitResponseStart(strictEntity(cth, input, bodyStart, cl))
-                setCompletionHandling(HttpMessageParser.CompletionOk)
-                emit(MessageEnd)
-                startNewMessage(input, bodyStart + cl)
-              } else {
-                emitResponseStart(defaultEntity(cth, contentLength))
-                parseFixedLengthBody(contentLength, closeAfterResponseCompletion)(input, bodyStart)
-              }
-            case None =>
-              emitResponseStart {
-                StreamedEntityCreator { entityParts =>
-                  val data = entityParts.collect { case EntityPart(bytes) => bytes }
-                  HttpEntity.CloseDelimited(contentType(cth), data)
+        case _ =>
+          if (!isChunked) {
+            clh match {
+              case Some(`Content-Length`(contentLength)) =>
+                if (contentLength == 0) finishEmptyResponse()
+                else if (contentLength <= input.size - bodyStart) {
+                  val cl = contentLength.toInt
+                  emitResponseStart(strictEntity(cth, input, bodyStart, cl))
+                  setCompletionHandling(HttpMessageParser.CompletionOk)
+                  emit(MessageEnd)
+                  startNewMessage(input, bodyStart + cl)
+                } else {
+                  emitResponseStart(defaultEntity(cth, contentLength))
+                  parseFixedLengthBody(contentLength, closeAfterResponseCompletion)(input, bodyStart)
                 }
-              }
-              setCompletionHandling(HttpMessageParser.CompletionOk)
-              parseToCloseBody(input, bodyStart, totalBytesRead = 0)
+              case None =>
+                emitResponseStart {
+                  StreamedEntityCreator { entityParts =>
+                    val data = entityParts.collect { case EntityPart(bytes) => bytes }
+                    HttpEntity.CloseDelimited(contentType(cth), data)
+                  }
+                }
+                setCompletionHandling(HttpMessageParser.CompletionOk)
+                parseToCloseBody(input, bodyStart, totalBytesRead = 0)
+            }
+          } else {
+            if (clh.isEmpty) {
+              emitResponseStart(chunkedEntity(cth), headers)
```

Comment:
Not 100% sure about just passing the headers here, vs addTransferEncodingWithChunkedPeeled like before, but it's now signalled by the boolean so I _think_ it is right?

Same logic in the `HttpRequestParser`

Reply:
Yes, we don't provide custom `Transfer-Encoding` any more, so :+1:.